### ISSUE TITLE
TruffleHog only finds last secret when multiple CustomRegex secrets present in same file

### ISSUE THREAD

**Issue by jlegarreta** [trufflesecurity/trufflehog#2113]
### Description
When there are multiple custom detectors defined, secrets matching more than one custom detector are found in the same file, and the secrets are not too far away from one another, only the last secret is found.

### TruffleHog Version

3.62.1
### Trace Output

With "bar" appearing last in secret file:
https://gist.github.com/jlegarreta/364cd3001184a372bb69284804719272

With "foo" appearing last in secret file:
https://gist.github.com/jlegarreta/8f6cc16e9061900bfca8a8724fba639c
### Expected Behavior

There should be two results, one for "foo" and one for "bar".

### Actual Behavior

There is only a result for "bar"

### Steps to Reproduce

```
echo foo >secret.txt
echo bar >>secret.txt
cat >config.yaml <<EOF
---
detectors:
  - name: Foo
    keywords:
      - foo
    regex:
      secret: foo
  - name: Bar
    keywords:
      - bar
    regex:
      secret: bar
EOF
trufflehog filesystem --config config.yaml --include-detectors=CustomRegex --no-update secret.txt
```
#### Output
```
Found unverified result 🐷🔑❓
Detector Type: CustomRegex
Decoder Type: PLAIN
Raw result: bar
Name: Bar
File: secret.txt
Line: 1
```
## Environment
 * OS: MacOS
 * Version: 12.7.1

**Comment by codenoid** [comment]
![image](https://github.com/trufflesecurity/trufflehog/assets/14269809/8e9c179e-3730-4471-8090-b28fd858fe01)

It seems just fine on windows

**Comment by zricethezav** [comment]
Thanks for opening this issue @jlegarreta, taking a look

**Comment by zricethezav** [comment]
@jlegarreta just cut a new release which has the fix https://github.com/trufflesecurity/trufflehog/pull/2065. Try the latest release and see if that fixes the issue

**Comment by jlegarreta** [comment]
@zricethezav I just upgraded to the latest release 3.63.0, and can confirm that this resolves the issue.

> ![image](https://user-images.githubusercontent.com/14269809/283793969-8e9c179e-3730-4471-8090-b28fd858fe01.png)
> 
> It seems just fine on windows

I should have included the flag `--no-update` in the trufflehog command in my steps to replicate (I added it now). Looks like they just pushed the fix before you tested this out, so you were probably using version 3.63.0. Thanks for testing though!
